lib/git-process.ts

```diff
@@ -311,3 +348,56 @@ function ignoreClosedInputStream(process: ChildProcess) {
     }
   })
 }
+
+export enum GitTaskCancelResult {
+  successfulCancel,
+  processAlreadyEnded,
+  noProcessIdDefined,
+  failedToCancel
+}
+
+/** This interface represents a git task (process). */
+export interface IGitTask {
+  /** Result of the git process. */
+  readonly result: Promise<IGitResult>
+  /** Allows to cancel the process if it's running. Returns true if the process was killed. */
+  readonly cancel: () => Promise<GitTaskCancelResult>
+  readonly updateProcessEnded: () => void
```

Comment:
I think this shouldn't be here, it should be part of the "hidden" interface (i.e. it should belong to `GitTask` only).
```suggestion
```